src/neml2/models/SymR2Invariant.cxx

```diff
@@ -71,7 +71,7 @@ SymR2Invariant::set_value(const LabeledVector & in,
   else if (_type == "I2")
   {
     if (out)
-      out->set((A.tr() * A.tr() - A.inner(A)) / 2, invariant);
+      out->set((A.tr() * A.tr() - A.inner(A)) / 2.0, invariant);
```

Comment:
Do we want similar override methods for `int`?  Maybe something to think about later (with macros...)

Reply:
yes with macros